src/main/java/com/github/pgasync/impl/PgConnectionPool.java

```diff
@@ -178,8 +178,7 @@ public void close() throws Exception {
                     lock.unlock();
                 }
             }
-        }).flatMap(conn -> validator.call(conn).doOnError(err -> release(conn)))
-                .retry(poolSize + 1);
+        }).flatMap(conn -> validator.call(conn).doOnError(err -> release(conn)));
```

Comment:
Is removing the retry call intentional? Why?

Reply:
it was. retry logic looks like a separate concern to me. for a heavy analytic query which timeout you do not want to retry it. 

for short simple query for sure.

in summary this needs to be revisited.